src/Behat/Config/Suite.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Behat\Config;
+
+final class Suite implements SuiteConfigInterface
+{
+    public function __construct(
```

Comment:
I don't like that `profile` needs to be passed as a parameter here. This means that if you have several profiles, in any which is not the default you need to be passing the profile name everywhere. I think that it would be best if we had a `withProfile()` function which accepted a `Profile` item (or its corresponding interface) and then this Profile class is the one that would have a `withSuite()` function. Adding the `withProfile` function and the `Profile` class would probably be best done in a separate PR that would be merged before this one

Reply:
Ok, let's go for it.